src/amp-story-player/amp-story-entry-point/amp-story-entry-point-impl.js

```diff
@@ -84,6 +113,170 @@ export class AmpStoryEntryPoint {
     shadowRoot.appendChild(this.rootEl_);
   }
 
+  /** @private */
+  initializeCarousel_() {
+    const style = this.element_.getAttribute('entry-point-style');
+    const styleClass = !Object.values(STYLE_TYPES).includes(style)
+      ? STYLE_CLASSES['rectangle']
+      : STYLE_CLASSES[style];
+
+    this.element_.classList.add(styleClass);
+
+    const cardsContainer = this.doc_.createElement('div');
+    cardsContainer.classList.add('entry-points');
+    this.rootEl_.append(cardsContainer);
+
+    for (let i = 0; i < this.stories_.length; i++) {
+      const story = this.stories_[i];
+      const src = story.getAttribute('data-poster-portrait-src');
+
+      const cardContainer = this.doc_.createElement('div');
+      cardContainer.classList.add('entry-point-card-container');
+      cardsContainer.append(cardContainer);
+
+      if (styleClass !== 'circular-entry-point') {
+        this.initializeCardTitle_(cardContainer, story);
+        this.initializeCardLogo_(cardContainer);
+      }
+
+      this.initializeCard_(cardContainer, src, styleClass, story.href, i);
+    }
+  }
+
+  /**
+   * Initializes the card's title to be displayed.
+   * @param {Element} cardContainer
+   * @param {HTMLAnchorElement} story
+   */
+  initializeCardTitle_(cardContainer, story) {
+    const title = this.doc_.createElement('span');
+    title.classList.add('entry-point-card-headline');
+    title.textContent = story.querySelector('span').textContent;
+    cardContainer.append(title);
+  }
+
+  /**
+   * Initializes the logo to be displayed.
+   * @param {Element} cardContainer
+   */
+  initializeCardLogo_(cardContainer) {
+    const src = this.element_.getAttribute('logo');
+    if (!src) {
+      return;
+    }
+
+    const logo = this.doc_.createElement('img');
+    logo.classList.add('entry-point-card-logo');
+    logo.src = src;
+    cardContainer.append(logo);
+  }
+
+  /**
+   * Initializes an image element to be displayed.
+   * @param {Element} cardContainer
+   * @param {string} src
+   * @param {string} styleClass
+   * @param {string} href
+   * @param {number} index
+   * @private
+   */
+  initializeCard_(cardContainer, src, styleClass, href, index) {
+    const card = this.doc_.createElement('img');
+    card.src = src;
+    card.classList.add(styleClass);
+    card.classList.add('entry-point-card-img');
+    this.setCardWidthHeight_(card, styleClass);
+    this.onCardClick_(card, href, index);
+    cardContainer.append(card);
+  }
+
+  /**
+   * Sets the correct height and width for poster based on style.
+   * @param {HTMLImageElement} card
+   * @param {string} styleClass
+   * @private
+   */
+  setCardWidthHeight_(card, styleClass) {
+    card.height = 233;
```

Comment:
This currently hard codes the width and height of the cards, but should be updated later on to allow for different sizes.